src/SmartComponents/BaselinesPage/DeleteBaselinesModal/DeleteBaselinesModal.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Modal, Button } from '@patternfly/react-core';
+
+import { baselinesTableActions } from '../../BaselinesTable/redux/index';
+
+class DeleteBaselinesModal extends Component {
+    constructor(props) {
+        super(props);
+        this.deleteBaselines = this.deleteBaselines.bind(this);
+
+        this.state = {
+            modalOpened: this.props.modalOpened
+        };
+
+        this.toggleModal = () => {
+            const { modalOpened } = this.state;
+
+            this.setState({ modalOpened: !modalOpened });
+        };
+    }
+
+    async deleteBaselines() {
+        const { clearSelectedBaselines, deleteSelectedBaselines, selectedBaselineIds, fetchBaselines, baselineId } = this.props;
+        let apiBody;
+
+        /*eslint-disable camelcase*/
+        if (baselineId) {
+            apiBody = { data: { baseline_ids: [ baselineId ]}};
+        } else {
+            apiBody = { data: { baseline_ids: selectedBaselineIds }};
+        }
+        /*eslint-enable camelcase*/
+
+        this.toggleModal();
+        clearSelectedBaselines();
+        await deleteSelectedBaselines(apiBody);
+        fetchBaselines();
+    }
+
+    render() {
+        const { modalOpened } = this.state;
+
+        return (
+            <Modal
+                title="Delete baselines"
+                className="small-modal-body"
+                isOpen={ modalOpened }
+                onClose={ this.toggleModal }
+                width="auto"
+                isFooterLeftAligned
+                actions = { [
+                    <Button
+                        key="confirm"
+                        variant="danger"
+                        onClick={ this.deleteBaselines }
+                    >
+                    Delete Baselines
+                    </Button>,
+                    <Button
+                        key="cancel"
+                        variant="secondary"
+                        onClick={ this.toggleModal }
+                    >
+                    Cancel
+                    </Button>
+                ] }
+            >
+                Deleting a baseline is permanent and cannot be undone.
+            </Modal>
+        );
+    }
+}
+
+DeleteBaselinesModal.propTypes = {
+    modalOpened: PropTypes.bool,
+    clearSelectedBaselines: PropTypes.func,
+    selectedBaselineIds: PropTypes.array,
+    deleteSelectedBaselines: PropTypes.func,
+    fetchBaselines: PropTypes.func
```

Comment:
This modal ends up having to know a lot about the inner workings of the baselines page and table. Could it be simplified to be passed a function that handles all of the deleting, clearing, knowledge of what is selected and then fetching?